integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/transaction/TransactionMediatorTestCase.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2017 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.esb.mediator.test.transaction;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.test.utils.http.client.HttpRequestUtil;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+
+import java.net.URL;
+
+/**
+ * Test the actions of Transaction Mediator including new, commit, rollback, fault-if-no-tx, use-existing-or-new, etc.
+ * <p>
+ * Perform different transactions and test for final indexes of the database entries
+ */
+public class TransactionMediatorTestCase extends ESBIntegrationTest {
+    private static final String API_URL = "http://localhost:8480/transaction-test/";
+    private static final String INIT_CONTEXT = "init";
+    private static final String COMMIT_CONTEXT = "commit";
+    private static final String ROLLBACK_CONTEXT = "rollback";
+    private static final String TEST_CONTEXT = "test";
+    private static final String CLEANUP_CONTEXT = "cleanup";
+    private static final String FAULT_NOTX_CONTEXT = "fault-if-no-tx";
+    private static final String USE_EXISTING_NEW_CONTEXT = "use-existing-or-new";
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        HttpRequestUtil.doPost(new URL(API_URL + INIT_CONTEXT), "");
```

Comment:
Why are we performing test work here? 
Do only environment initialization work here

Reply:
We need to initialize database(create tables and relevant db entries) before the following tests. Is there any other common place to perform those actions?

Reply:
Ah .. if it is the intention, then this is fine. Shall we add a comment here expressing the intention of this API call

Reply:
Sure, will add a comment.